### ISSUE TITLE
dotnet-install results in "Killed: 9" when updating a signed build on MacOs

### ISSUE THREAD

**Issue by ericstj** [dotnet/install-scripts#232]
Using CP here https://github.com/dotnet/install-scripts/blob/0aa4951a546b42c6a4ff5ed553065ddf0a494098/src/dotnet-install.sh#L838

Is overwriting `dotnet`.

When `dotnet` is a signed file, and has previously run, Mac OS is caching the signature.  When that file is replaced "in place" as is done with `cp` it causes subsequent execution of `dotnet` to fail with `Killed: 9`

I've reported this to Apple, but the scripts should avoid this pattern if possible.  I believe the MacOs installer first moves files out of the way, then deletes them, before copying a new file.

**Comment by ericstj** [comment]
@joeloff looped me into this happening in the SDK repo where it looks like they weren't passing that flag.

**Comment by joeloff** [comment]
Good to know about that flag. I'll update the SDK PR

**Comment by bekir-ozturk** [comment]
We have synced with @joeloff and it appears that `SkipNonVersionedFiles` switch helps in this case.
I am closing this issue.

**Comment by ericstj** [comment]
External customers are also hitting this.  IMO we shouldn't have an install script which by default overwrites and downgrades files.  I don't suppose we could add version detection to the scripts?  We have means for defining this on windows (filever) and unix (strings).

**Comment by bekir-ozturk** [comment]
Since these files are considered "unversioned", shouldn't the difference between them negligible? Is there a noticeable difference between invoking `dotnet.exe` generated by SDK 3.1 vs SDK 7.0?
I believe the only thing that the dotnet executable is doing today is to determine the sdk version to be used and redirect the command to the versioned dotnet.dll.

**Comment by rolfbjarne** [comment]
This started happening on most (all?) of our macOS 12.0 (Ventura) bots just after .NET 8.0.406 was released.

We're simply doing this:

```yaml
- task: UseDotNet@2
  inputs:
    version: 8.x
```

Running this on the bots fixed the problem:

```
cp /Users/builder/azdo/_work/_tool/dotnet/dotnet /Users/builder/azdo/_work/_tool/dotnet/dotnet2
mv /Users/builder/azdo/_work/_tool/dotnet/dotnet2 /Users/builder/azdo/_work/_tool/dotnet/dotnet 
```

FWIW this _didn't_ happen on our macOS 13.0-15.0 bots, so this may have been fixed by Apple in macOS 13.0+.
